qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -414,10 +424,54 @@ def open_file(self):
             return
         self.editor.open_file_in_new_tab(filepath)
 
-    def save_file(self):
-        # todo: how should this interact with project saving and workspaces when they are implemented?
+    def save_script(self):
         self.editor.save_current_file()
 
+    def save_project(self):
+        if self.last_project_location is None:
```

Comment:
I'm wondering if we are missing a concept here. How about having a `Project` type. `MainWindow`could then create an instance of a project class on startup that then has methods `save`/`load`/last_location etc. It would be easier to unit test this behaviour then.

Reply:
Agreed and Implemented